### ISSUE TITLE
Epoch-wise losses

### ISSUE THREAD

**Issue by danielkelshaw** [HamzaSardar/PyCoarseNet#6]
As per the code below, the losses are being appended to a list at each training step. This will quickly fall down if/when you switch to using a `DataLoader` (#5) and run through with batches. Consider keeping an epoch-wise loss which is then normaliser wrt. the number of elements in the train/validation set.

https://github.com/HamzaSardar/PyCoarseNet/blob/47baebc46cdd1bf1b6ce11ff8abcdabf9e6129fc/pycoarsenet/training/model_setup.py#L42
